array.py

```diff
@@ -95,33 +95,31 @@ class ArrayIsh(Protocol):
     shape: tuple[int, ...]
     strides: tuple[int, ...]
     dtype: np.dtype[Any]
+    bytes: int
 
 
 class ArrayFlags:
-    f_contiguous: bool
-    c_contiguous: bool
-    forc: bool
```

Comment:
As a matter of style, I actually kind of like these attribute declarations, and I think the overall code is more readable with them, compared to having the same information buried deep in the constructor. Case-by-case judgment is certainly appropriate here (I don't want to totally disallow one style or the other), but I generally feel that leaning towards including explicit attribute annotations outside the constructor is a net positive for readability.

I won't let that stand in the way of merging this though.

Reply:
Are they the same? I think I saw `pyright` complaining about `self.shape` in something like
```
    @property
    def ndim(self) -> int:
        return len(self.shape)
```
Might have gotten confused about class vs instance attributes (can't recall the exact message). Also, I know Sphinx works with `self.something: Type = something`. Does it also pick up the annotations in this style?

Reply:
> For what it's worth, I also prefer this since it has all the type annotations nicely in one place. I think I started adding them in the constructor for some simple classes and then just stuck with that for consistency..

Cool, sounds like we agree style-wise then. No need to rip up existing usage of in-constructor declaration, but I think it's worth it to bias in favor of class-body declarations going forward.